Framework/Algorithms/src/SpecularReflectionPositionCorrect2.cpp

```diff
@@ -40,14 +40,14 @@ double offsetAngleFromCentre(const MatrixWorkspace &ws, const double l2,
                              const double linePosition,
                              const double pixelSize) {
   const auto &spectrumInfo = ws.spectrumInfo();
-  const size_t nSpec = spectrumInfo.size();
-  if (linePosition + 1. > static_cast<double>(nSpec)) {
+  const size_t nSpec = spectrumInfo.size() - 1;
```

Comment:
This should probably be called `maxWorkspaceIndex`. Clearly it is not the number of spectra anymore because of the `- 1`.